docs/guide/snaps-development-guide.md

```diff
@@ -1,47 +1,290 @@
 
-For present purposes, the important thing to understand is that, because SES is effectively a subset of JavaScript, you may write - or more likely, import - code that does not execute under SES.
-[`@metamask/snaps-cli`](https://www.npmjs.com/package/@metamask/snaps-cli) will attempt to fix some such issues for you via the `mm-snap build` command, but it can't fix them all.
+#### Secure EcmaScript (SES)
+
+Secure EcmaScript, or [SES](https://github.com/endojs/endo/tree/master/packages/ses), is effectively a subset of the JavaScript language designed to enable mutually suspicious programs to execute in the same JavaScript process (or more accurately, the same [realm](https://tc39.es/ecma262/#realm)).
+You can think of it as a more severe form of [strict mode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode).
+
+### Fixing Build / Eval Issues
+
+Because SES adds additional restrictions on the JavaScript runtime on top of strict mode, code that executes normally under strict mode may not do so under SES.
+`mm-snap build` will by default attempt to execute your snap in a stubbed SES environment.
+You can also disable this behavior and run the evaluation step separately using `mm-snap eval`.
+If an error is thrown during this step, it is likely due to a SES incompatibility, and you have to fix the issues manually.
+In our experience, these incompatibilities tend to occur in dependencies.
+
+While the errors you get from SES may seem scary, they're usually not that hard to fix.
+The actual file, function, and variable names in the `mm-snap eval` error stack trace may not make a lot of sense to you, but the line numbers should correspond to your snap bundle file.
+In this way, you can identify if the error is due to your code or one of your dependencies.
+If the problem is in a dependency, you can try a different version or to fix the issue locally by using tools such as [`patch-package`](https://npmjs.com/package/patch-package) or modifying the snap bundle file directly.
+
+To give you an idea of a common error and how to fix it, "sloppily" declared variables (i.e. assigning to a new variable without an explicit variable declaration) are forbidden in strict mode, and therefore in SES as well.
+If you get an error during the `eval` step that says something like `variableName is not defined`, simply prepending `var variableName;` to your snap bundle may solve the problem.
+(This actually happened so frequently with [Babel's](https://babeljs.io/) `regeneratorRuntime` that `mm-snap build` automatically handles that one.)
+
+::: warning Did you modify the snap bundle after running `mm-snap build`?
+Don't forget to run `mm-snap manifest --fix` if you modified your snap bundle after building.
+:::
+
+If you run into a build or eval issue that you can't solve on your own, please create an issue on the [MetaMask/snaps-skunkworks](https://github.com/MetaMask/snaps-skunkworks) repository.
```

Comment:
Skunkworks repo? Are we deprecating that?

Reply:
Not any time soon! Some packages graduate out of there, but others stick around.